benchmarks/conv.dx

```diff
@@ -26,8 +26,13 @@ def conv_1d {d1 d2} (kernel: (Fin d1)=>(Fin d2)=>Float)
 
 def conv {n c h w} (kernel: (Fin n)=>(Fin c)=>(Fin h)=>(Fin w)=>Float)
   (size: Int) : (Fin n)=>(Fin c)=>(Fin h)=>(Fin w)=>Float =
-  for n' c'.
-    conv_1d kernel.n'.c' (unsafe_i_to_n size)
+  for n' c'. conv_1d kernel.n'.c' (unsafe_i_to_n size)
+
+def conv_spec {n c h w} (kernel: (Fin n)=>(Fin c)=>(Fin h)=>(Fin w)=>Float)
+  (size: Int) : (Fin n)=>(Fin c)=>(Fin h)=>(Fin w)=>Float =
+  if size == 3
+    then conv kernel 3
+    else conv kernel size
```

Comment:
This case seems a little ad-hoc and will affect the Dex-to-Dex benchmark too (the baseline does not have that conditional!)

Reply:
The Dex-to-Dex benchmark circumvents the conditional by calling `conv`, not `conv_spec`.

The conditional is there because, as of commit dfef51635df807a59b2cddc748fffe4872fa4575, it speeds the Jax-calls-Dex version of this benchmark up from 21 ms to 0.23 ms (on my laptop at some unknown load level), i.e., about 100x.  I presume this is because that lets it hit the successfully specialized code path; though perhaps there Dougal's idea about merging lowering paths will fix this.

Reply:
Ah I see now. Fine, makes sense.